src/ApiService/ApiService/OneFuzzTypes/Model.cs

```diff
@@ -471,19 +475,33 @@ public class NotificationTemplateConverter : JsonConverter<NotificationTemplate>
     public override NotificationTemplate? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options) {
         using var templateJson = JsonDocument.ParseValue(ref reader);
         try {
-            return templateJson.Deserialize<AdoTemplate>(options);
-        } catch (JsonException) {
+            return ValidateDeserialization(templateJson.Deserialize<AdoTemplate>(options));
+        } catch (Exception ex) when (
+            ex is JsonException
```

Comment:
log exceptions ?

Reply:
The exceptions are just a mechanism to "fall-through" until we deserialize to the right type, that's why they're always caught. The only exception that's important is the last one because it indicates we weren't able to deserialize to any of our known types. Also, logging isn't available here, it would have to be done after serialization (which we already do).

I improved the exception message to be clearer in the case where none of the type deserialize.